pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -67,17 +66,35 @@ func (reconciler *Reconciler) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	// Fetch the KieApp instance
-	instance := &api.KieApp{}
+	instance := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      request.Name,
+			Namespace: request.Namespace,
+		},
+	}
+	var crDeleted bool
 	err := reconciler.Service.Get(context.TODO(), request.NamespacedName, instance)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			return reconcile.Result{}, nil
+			log.Info("Resource is being deleted. Reconcile for deletion.")
+			crDeleted = true
+		} else {
+			// Error reading the object - requeue the request.
+			reconciler.setFailedStatus(instance, api.UnknownReason, err)
+			return reconcile.Result{}, err
 		}
-		// Error reading the object - requeue the request.
-		reconciler.setFailedStatus(instance, api.UnknownReason, err)
+	}
+	requestedResources := []resource.KubernetesResource{}
+
+	if crDeleted {
+		deployed, err := reconciler.getDeployedResources(instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		_, err = reconciler.reconcileResources(instance, requestedResources, deployed)
```

Comment:
Move above 6-7 lines to if IsNotFound

Reply:
I think you might be able to just pass nil for requestedResources and it will default to empty slice... if I recall correctly